io/multiwriter.go

```diff
@@ -0,0 +1,56 @@
+package io
+
+import (
+	"errors"
+	"io"
+	"sync"
+)
+
+var ErrShortWrite = errors.New("short write")
+
+type asyncMultiWriter struct {
+	writers []io.Writer
+}
+
+// AsyncMultiWriter creates a writer that duplicates its writes to all the
+// provided writers asynchronous
+func AsyncMultiWriter(writers ...io.Writer) io.Writer {
+	w := make([]io.Writer, len(writers))
+	copy(w, writers)
+	return &asyncMultiWriter{w}
+}
+
+// Writes data asynchronously to each writer and waits for all of them to complete.
+// In case of an error, the writing will not complete.
+func (t *asyncMultiWriter) Write(p []byte) (int, error) {
+	var wg sync.WaitGroup
+	wg.Add(len(t.writers))
+	errChannel := make(chan error)
```

Comment:
I think you should initialize it with the number of writers so that if an error is encountered and already written, there won't be a deadlock
```suggestion
	errChannel := make(chan error, len(t.writers))
```